docs/concepts/ingress-classes.md

```diff
@@ -0,0 +1,181 @@
+
+Because they create Kong configuration indenpendent of any other resources,
+directly-translated resources require an ingress class, and their class must
+match the class configured for the controller. Referenced resources do not
+require a class, but must be referenced by a directly-translated resource
+matches the controller.
+
+### Adding class information to resources
+
+Most resources use a [kubernetes.io/ingress-class annoration][class-annotation]
+to indicate their class. There are several exceptions:
+
+- v1 Ingress resources have a dedicated `class` field.
+- Knative Services [use the class specifed][knative-class] by the
+  `ingress.class` key of the Knative installation's `config-network` ConfigMap.
+  You can optionally [override this on a per-Service basis][knative-override]
+  by adding a `networking.knative.dev/ingress.class` annotation to the Service.
+
+### Enabling support for classless resources
+
+Specifying a class is optional for some resources. Although specifying a class
+is recommended, you can instruct the controller to process resources without a
+class annotation using flags:
+
+- `--process-classless-ingress-v1beta1` instructs the controller to translate
+  v1beta1 Ingress resources with no class annotation.
+- `--process-classless-kong-consumer` instructs the controller to translate
+  KongConsumer resources with no class annotation.
+
+These flags are primarily intended for compatibility with older configuration
+(older versions of the Kong Ingress Controller had less strict class
+requirements, and it was common to omit class annotations). If you are creating
+new configuration and do not have older configuration without class
+annotations, recommended best practice is to add class information to Ingress
+and KongConsumer resources and not set the above flags. Doing so avoids
+accidentally creating duplicate configuration in other ingress controller
+instances.
+
+These flags do not _ignore_ `ingress.class` annotations: they allow resources
+with no such annotation, but will not allow resource that have a non-matching
+`ingress.class` annotation.
+
+## When to use a custom class
+
+Using the default `kong` class is fine for simpler deployments, where only one
+Kong Ingress Controller instance is running in a cluster. Changing the class is
+typical when:
+
+- You install multiple Kong environments in one cluster to handle different
+  types of ingress traffic, e.g. when using separate Kong instances to handle
+  traffic on internal and external load balancers, or deploying different types
+  of non-production environments in a single test cluster.
+- You install multiple controller instances that create configuration in
+  different Kong workspaces (using the `--kong-workspace` flag).
+
+## Legacy behavior
+
+This overview covers behavior in Kong Ingress Controller version 0.10.0 onward.
+Earlier versions had a special case for the default class and a bug affecting
```

Comment:
Unnecessary IMO; the explanation here is sufficient, and the original bug. https://github.com/Kong/kubernetes-ingress-controller/issues/735, would confuse readers because they must navigate away from the document, interrupting flow, and because the original bug description is both narrower than the total scope of the problem (it does not only affect KongPlugins) and discusses the technical underpinnings of the issue rather than its user-facing effects. Dismissed.